### PR TITLE
Tao 480 improve dac user interface

### PR DESCRIPTION
**Pull request by jsconan**
https://oat-sa.atlassian.net/browse/TAO-480

Sub tasks covered:
- https://oat-sa.atlassian.net/browse/TAO-461
- https://oat-sa.atlassian.net/browse/TAO-738

Improve the user interface to grant or remove permissions of users or roles from a resource in the Tao back-office by:
 - Allow existing roles to be removed before adding new ones, giving an error message on submit if the roles defined would leave the resource unmanageable.
 - Replace the list of users and roles by an auto-complete field that uses the search to find the desired role/user. If a user tries to read a role or user that is already in the list, highlight it.

Requires:
- https://github.com/oat-sa/tao-core/pull/328
- https://github.com/oat-sa/tao-core/pull/330

### CODE REVIEW

**Comment by krampstudio** [comment]
I'll review client side part and @jbout will review php side.

File: views/scss/admin.scss [hunk 1]
```diff
@@ -2,7 +2,45 @@
...
 .permission-container {
-    table td {
-        min-width: 130px;
+    .grid-container {
+        margin-top: 40px;
+        margin-bottom: 20px;
+    }
+
+    .msg-edit-area {
+        border: 1px $uiGeneralContentBorder solid;
```
Comment (krampstudio) [inline views/scss/admin.scss]: Isn't this what `@simple-border` does

File: views/scss/admin.scss [hunk 2]
```diff
@@ -2,7 +2,45 @@
...
+        background: $canvas;
+        padding: 20px 20px 10px 20px;
+        @include simple-border();
+
+        .icon-add {
+            font-size: 2rem !important;
```
Comment (krampstudio) [inline views/scss/admin.scss]: There's a `font-size` (check for the exact name) mixin that you should use instead

Comment (jsconan) [inline views/scss/admin.scss]: I replaced the classic rules by the related mixins

2015-06-02 10:10 GMT+02:00 Bertrand CHEVRIER <notifications@github.com>:

> In views/scss/admin.scss
> <https://github.com/oat-sa/extension-tao-dac-simple/pull/15#discussion_r31500393>
> :
>
> > @@ -2,7 +2,45 @@
> >  @import "inc/buttons";
> >
> >  .permission-container {
> > -    table td {
> > -        min-width: 130px;
> > +    .grid-container {
> > +        margin-top: 40px;
> > +        margin-bottom: 20px;
> > +    }
> > +
> > +    .msg-edit-area {
> > +        border: 1px $uiGeneralContentBorder solid;
> > +        background: $canvas;
> > +        padding: 20px 20px 10px 20px;
> > +        @include simple-border();
> > +
> > +        .icon-add {
> > +            font-size: 2rem !important;
>
> There's a font-size (check for the exact name) mixin that you should use
> instead
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/oat-sa/extension-tao-dac-simple/pull/15/files#r31500393>
> .
>



-- 




*Jean-Sébastien CONAN*
Application Architect

•   Email:   jean-sebastien.conan@vesperiagroup.com
•   GSM: (+352) 621 320 343

*Vesperia Group*
148, Route d'Arlon
L-8010 Strassen
Luxembourg

http://www.vesperiagroup.com/

**Comment by krampstudio** [comment]
From my side everything seems correct. Just one question left : 
Shouldn't the `recursive` checkbox be hidden/disabled when we are on an instance and enabled only on classes?

Reassigning it @jbout  for review

**Comment by jsconan** [comment]
I changed the way the `recursive` checkbox is displayed : if the resource is not a class, the checkbox is not displayed

**Comment by jbout** [comment]
update script to be provided

**Pull request merged**